### PR TITLE
Backfill tests for 4xx or 5xx status code

### PR DESCRIPTION
**Pull request by mkante**
This PR is a follow up of issue #189 
I added more tests to support feature/issue-179

### CODE REVIEW

File: packages/zipkin-instrumentation-request-promise/test/integrationTest.js [hunk 1]
```diff
@@ -262,7 +275,91 @@ describe('request instrumentation - integration test', () => {
...
+          expect(annotations[6].annotation.annotationType).to.equal('ClientRecv');
+
+          expect(annotations[7]).to.be.undefined; // eslint-disable-line no-unused-expressions
+          done();
+        });
+      });
+    });
+  });
+
+  it('should report when service returns 500', done => {
```
Comment (jcchavezs) [inline packages/zipkin-instrumentation-request-promise/test/integrationTest.js]: @mkante I believe we should add an error tag on status codes 4xx and 5xx. We do so in other http client libraries: https://github.com/openzipkin/zipkin-go/blob/master/middleware/http/client.go#L119 and https://github.com/jcchavezs/zipkin-instrumentation-guzzle/blob/master/src/ZipkinGuzzle/Middleware.php#L61

I am surprised we don't do this in https://github.com/openzipkin/zipkin-js/blob/master/packages/zipkin-instrumentation-request/test/integrationTest.js#L231.

Any input @adriancole.

**Comment by adriancole** [comment]
looks like an oversight to me

File: packages/zipkin-instrumentation-request-promise/test/integrationTest.js (hunk 1 above)
Comment (mkante) [reply packages/zipkin-instrumentation-request-promise/test/integrationTest.js]: Hm I see want you mean

Comment (jcchavezs) [reply packages/zipkin-instrumentation-request-promise/test/integrationTest.js]: Sounds good to me. Any opinion on this @adriancole?

Comment (adriancole) [reply packages/zipkin-instrumentation-request-promise/test/integrationTest.js]: usually also check <200 (though this should be not usual :P)

Comment (jcchavezs) [reply packages/zipkin-instrumentation-request-promise/test/integrationTest.js]: @mkante can you please add that change as well?

Comment (jcchavezs) [reply packages/zipkin-instrumentation-request-promise/test/integrationTest.js]: On this topic I just opened this PR as an attempt to uniformize this errors: https://github.com/openzipkin/zipkin-js/pull/261 cc @mkante @adriancole

Comment (mkante) [reply packages/zipkin-instrumentation-request-promise/test/integrationTest.js]: Thanks I will look into this weekend

**Comment by jcchavezs** [comment]
@mkante can we rebase this PR?

**Comment by mkante** [comment]
@jcchavezs Yep no problem, I will do it tonight

**Comment by jcchavezs** [comment]
Awesome!

Den tor. 13. sep. 2018, 20:30 skrev mkante <notifications@github.com>:

**Comment by jcchavezs** [comment]
:D This is great. Thanks a lot @mkante!

**Pull request merged**